magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,349 @@
+        trans = trans ** (1 / np.cos(zenith))
+        nlayers = correction_params.get("number_of_layers")
+        Hcl = np.linspace(Hc, Hc + dHc, nlayers)  # position of each layer
+        transl = trans_height(Hcl, Hc, dHc, trans)  # transmissions of each layer
+        transl = np.append(transl, transl[-1])
+
+        alt_rad = np.deg2rad(dl1_params["alt"][index])
+        az_rad = np.deg2rad(dl1_params["az"][index])
+
+        impact = dl1_params["impact"][index] * u.m
+        psi = dl1_params["psi"][index] * u.deg
+        cog_x = (dl1_params["x"][index] * m2deg).value * u.deg
+        cog_y = (dl1_params["y"][index] * m2deg).value * u.deg
+
+        # Source position
+        pointing_altaz = SkyCoord(alt=alt_rad * u.rad, az=az_rad * u.rad, frame=AltAz())
+        telescope_pointing = SkyCoord(
+            alt=pointing_alt * u.rad,
+            az=pointing_az * u.rad,
+            frame=AltAz(),
+        )
+
+        tel_frame = TelescopeFrame(telescope_pointing=telescope_pointing)
+        tel = pointing_altaz.transform_to(tel_frame)
+
+        src_x = tel.fov_lat
+        src_y = tel.fov_lon
+
+        # Transform to Engineering camera
+        src_x, src_y = -src_y, -src_x
+        cog_x, cog_y = -cog_y, -cog_x
+
+        pix_x_tel = (camgeom[telid].pix_x * m2deg).to(u.deg)
+        pix_y_tel = (camgeom[telid].pix_y * m2deg).to(u.deg)
+
+        distance = np.abs(
+            (pix_y_tel - src_y) * np.cos(psi) + (pix_x_tel - src_x) * np.sin(psi)
+        )
+
+        d2_cog_src = (cog_x - src_x) ** 2 + (cog_y - src_y) ** 2
+        d2_cog_pix = (cog_x - pix_x_tel) ** 2 + (cog_y - pix_y_tel) ** 2
+        d2_src_pix = (src_x - pix_x_tel) ** 2 + (src_y - pix_y_tel) ** 2
+
+        distance[d2_cog_pix > d2_cog_src + d2_src_pix] = 0
+        dist_corr_layer = model2(impact, Hcl, zenith) * u.deg
+
+        ilayer = np.digitize(distance, dist_corr_layer)
+        trans_pixels = transl[ilayer]
+        if (trans_pixels == 0).any():
+            raise ValueError("trans_pixels must not contain any zero values")
+
+        inds_img = np.where(
+            (dl1_images["event_id"] == event_id)
+            & (dl1_images["tel_id"] == telid)
+            & (dl1_images["obs_id"] == obs_id)
+        )[0]
+
+        if len(inds_img) > 0:
+            for index_img in inds_img:
+                image = dl1_images["image"][index_img]
+                cleanmask = dl1_images["image_mask"][index_img]
+                peak_time = dl1_images["peak_time"][index_img]
+                image /= trans_pixels
+                corr_image = image.copy()
+                corr_image[~cleanmask] = 0
+
+                hillas_params = hillas_parameters(camgeom[telid], corr_image)
+                timing_params = timing_parameters(
+                    camgeom[telid], corr_image, peak_time, hillas_params
+                )
+                leakage_params = leakage_parameters(
+                    camgeom[telid], corr_image, signal_pixels
```

Comment:
this looks wrong. you pass here n_pixels, while the leakage calculation code requires cleaning mask: https://ctapipe.readthedocs.io/en/v0.19.3/api/ctapipe.image.leakage_parameters.html